embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/capture/crumbs/EmbraceBreadcrumbService.kt

```diff
@@ -24,13 +23,12 @@ import io.embrace.android.embracesdk.session.lifecycle.ActivityLifecycleListener
  * in server-side configuration. They are stored in an unbounded queue.
  */
 internal class EmbraceBreadcrumbService(
-    private val clock: Clock,
     private val configService: ConfigService,
     private val dataSourceModuleProvider: Provider<DataSourceModule?>
 ) : BreadcrumbService, ActivityLifecycleListener, MemoryCleanerListener {
 
-    override fun logView(screen: String?, timestamp: Long) {
-        dataSourceModuleProvider()?.viewDataSource?.dataSource?.changeView(screen, false)
+    override fun logView(screen: String?) {
```

Comment:
Do we not require the ability to specify a time in case the logging is delayed?

Reply:
we always use the current clock time, at least for this flow!

Reply:
So it wasn't passed in or was it just never actually used? If it's the latter, we may have a bug, albeit a subtle one

Reply:
Looking at the code, I think we should actually respect this and pass it down. We should use the time when the API method is called instead of when we are processing the event, so I think it's right for the RN call to pass it down - and we should probably do the same for the manual `startView` and `endView` calls as well